src/CMakeLists.txt

```diff
@@ -7,16 +7,28 @@ elseif(UNIX)
 endif()
 
 if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
-  set(CMAKE_Fortran_FLAGS "-g -fpp -free -xHOST -traceback -FR")
-  set(fortran_8_flags "-i8 -r8")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+  set(CMAKE_C_FLAGS_DEBUG "-g -traceback")
 elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
   set(CMAKE_C_FLAGS_DEBUG "-ggdb -Wall")
-  set(CMAKE_Fortran_FLAGS "-g -ffree-form -fbacktrace -funroll-loops")
+endif()
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -fpp -traceback -FR ${CMAKE_Fortran_FLAGS}")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-g -traceback")
+  set(fortran_8_flags "-i8 -r8")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
+  set(CMAKE_Fortran_FLAGS "-g -fbacktrace -funroll-loops ${CMAKE_Fortran_FLAGS}")
   set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall")
   set(fortran_8_flags "-fdefault-integer-8 -fdefault-real-8")
 endif()
 
-set(fortran_src baciof.f bafrio.f chk_endianc.f)
+if(${CMAKE_Fortran_COMPILER_ID} MATCHES "^(GNU)$" AND ${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -w -fallow-argument-mismatch")
+endif()
+
+set(fortran_src baciof.F90 bafrio.F90 chk_endianc.F90)
```

Comment:
Are there preprocessor directives in these?

Reply:
No

Reply:
I think only `bacio.F90` has `include` and that is fine.
The other 2 can be `.f90`

Reply:
We probably have diverging philosophies on this. My take is that preprocessing the file doesn't cost much and avoids potential changes in the future. But either way is ok for me.

Reply:
- Just looking at the file extension tells us that there is preprocessor involved.
- CMake processes those files first to produce `.f90` which are then compiled.

Reply:
That's actually a Fortran `Include` which doesn't use the pre-processor

Reply:
even better. remove the `-fpp` completely and turn those files to `.f90`